tests/quantization/autoawq/test_awq.py

```diff
@@ -0,0 +1,141 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tempfile
+import unittest
+
+from transformers import AutoModelForCausalLM, AutoTokenizer, AwqConfig
+from transformers.testing_utils import (
+    require_accelerate,
+    require_auto_awq,
+    require_torch_gpu,
+    slow,
+    torch_device,
+)
+
+
+@require_torch_gpu
+class AwqConfigTest(unittest.TestCase):
+    def test_wrong_backend(self):
+        """
+        Simple test that checks if a user passes a wrong backend an error is raised
+        """
+        # This should work fine
+        _ = AwqConfig(bits=4)
+
+        with self.assertRaises(ValueError):
+            AwqConfig(bits=4, backend="")
+
+        # LLMAWQ does not work on a T4
+        with self.assertRaises(ValueError):
+            AwqConfig(bits=4, backend="llm-awq")
+
+    def test_to_dict(self):
+        """
+        Simple test that checks if one uses a config and converts it to a dict, the dict is the same as the config object
+        """
+        quantization_config = AwqConfig(bits=4)
+        config_to_dict = quantization_config.to_dict()
+
+        for key in config_to_dict:
+            self.assertEqual(getattr(quantization_config, key), config_to_dict[key])
+
+    def test_from_dict(self):
+        """
+        Simple test that checks if one uses a dict and converts it to a config object, the config object is the same as the dict
+        """
+        dict = {"bits": 2, "zero_point": False, "backend": "autoawq"}
+        quantization_config = AwqConfig.from_dict(dict)
+
+        self.assertEqual(dict["bits"], quantization_config.bits)
+        self.assertEqual(dict["zero_point"], quantization_config.zero_point)
+        self.assertEqual(dict["backend"], quantization_config.backend)
+
+
+@slow
+@require_torch_gpu
+@require_auto_awq
+@require_accelerate
+class AwqTest(unittest.TestCase):
```

Comment:
Can you also test the multi-gpu setup `device_map="auto"` ?

Reply:
Done!